src/standalone/Soil/energy_hydrology.jl

```diff
@@ -332,42 +332,35 @@ function ClimaLand.make_compute_jacobian(model::EnergyHydrology{FT}) where {FT}
         # If the top BC is a `MoistureStateBC`, add the term from the top BC
         #  flux before applying divergence
         if haskey(p.soil, :dfluxBCdY)
-            dfluxBCdY = p.soil.dfluxBCdY
-            topBC_op = Operators.SetBoundaryOperator(
-                top = Operators.SetValue(dfluxBCdY),
+            dfluxBCdY_heat = p.soil.dfluxBCdY.heat
+            topBC_op_heat = Operators.SetBoundaryOperator(
+                top = Operators.SetValue(dfluxBCdY_heat),
+                bottom = Operators.SetValue(
+                    Geometry.Covariant3Vector(zero(FT)),
+                ),
+            )
+            dfluxBCdY_water = p.soil.dfluxBCdY.water
+            topBC_op_water = Operators.SetBoundaryOperator(
+                top = Operators.SetValue(dfluxBCdY_water),
                 bottom = Operators.SetValue(
                     Geometry.Covariant3Vector(zero(FT)),
                 ),
             )
-            # Add term from top boundary condition before applying divergence
-            # Note: need to pass 3D field on faces to `topBC_op`. Interpolating `K` to faces
-            #  for this is inefficient - we should find a better solution.
-            @. ∂ϑres∂ϑ =
-                -dtγ * (
-                    divf2c_matrix() ⋅ (
-                        MatrixFields.DiagonalMatrixRow(
-                            interpc2f_op(-p.soil.K),
-                        ) ⋅ gradc2f_matrix() ⋅ MatrixFields.DiagonalMatrixRow(
-                            ClimaLand.Soil.dψdϑ(
-                                hydrology_cm,
-                                Y.soil.ϑ_l,
-                                ν - Y.soil.θ_i, #ν_eff
-                                θ_r,
-                                S_s,
-                            ),
-                        ) + MatrixFields.LowerDiagonalMatrixRow(
-                            topBC_op(
-                                Geometry.Covariant3Vector(
-                                    zero(interpc2f_op(p.soil.K)),
-                                ),
-                            ),
-                        )
-                    )
-                ) - (I,)
         else
-            @. ∂ϑres∂ϑ =
-                -dtγ * (
-                    divf2c_matrix() ⋅
+            topBC_op_heat = Operators.SetBoundaryOperator(
+                top = Operators.SetValue(zero(FT)),
+                bottom = Operators.SetValue(
+                    Geometry.Covariant3Vector(zero(FT)),
+                ),
+            )
+            topBC_op_water = topBC_op_heat
+        end
+        # Add term from top boundary condition before applying divergence
+        # Note: need to pass 3D field on faces to `topBC_op`. Interpolating `K` to faces
```

Comment:
When you wrote "this is inefficient", why is that?

Reply:
@juliasloan25 do you recall what this was about?

Reply:
I think it's because we call `interpc2f_op(p.soil.K)` and immediately zero out the values, so we could use any field defined on faces. If we have a field that's already on faces we can avoid doing the extra interpolation.